### DISCUSSION TITLE
How does user_loader works in Flask-Login > 0.4.1

### DISCUSSION THREAD

**Discussion by saromba** [maxcountryman/flask-login#733] [Q&A]
When I update my project from Flask-Login 0.4.1 to 0.5.0 (or latest), the 'user_loader' decorator is no more working.
When later calling @roles_required('Read') I get always an Unauthorized Access...

```
in the app.py
...
user_manager = ApiUserManager(app.app, db, ApiUser, auth_object)
login_manager = user_manager.login_manager


@login_manager.user_loader
def load_user(user_id):
    try:
        return ApiUser.query.filter(ApiUser.id == user_id).first()
    except:
        return None

```
How does this work at the moment?

**Comment by davidism** [comment]
It continues to work as shown. You can refer to the docs for examples if you're unsure. It sounds like you have an unrelated error in your code, and have incorrectly identified Flask-Login as the cause. You might try taking out the bare `except` block to start debugging, as that's an antipattern in Python in general.

**Reply by saromba** [reply]
Ok thanks for your quick reply... I'm still trying to find my error.
Could it be that the user_id in the flask session has changed from "user_id" to "_user_id" ?
